File: BRACU ACM Student Chapter/Online Workshop on Programming Skills DAY 02 (Java) BRACU Number Theory/QNEWB.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
public class QNEWB {

    public static void main(String[] args) throws IOException {
        {
            BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
            int num = Integer.parseInt(input.readLine());
            boolean condition = false;
            int sum = 0;
            for (int i = 1; sum <= num; i++) {
                sum = sum + i;
                if (sum == num) condition = true;
            }
            System.out.println(condition ? "YES" : "NO");
        }
    }
}

File: BRACU ACM Student Chapter/Online Workshop on Programming Skills DAY 02 ( C++ ) BRACU  Number Theory/QNEWB.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int num, sum=0;
    cin >> num;
    bool condition = false;
    for(int i=1; sum<=num; i++)
    {
        sum+=i;
        if(sum==num) condition = true;
    }
    condition ? cout<<"YES" : cout<<"NO";
    return 0;
}
